src/libraries/System.Net.Sockets/tests/FunctionalTests/MulticastOptionTest.cs

```diff
@@ -26,8 +26,7 @@ public void MulticastOption_Group_Roundtrips()
             var option = new MulticastOption(IPAddress.Any);
             Assert.Same(IPAddress.Any, option.Group);
 
-            option.Group = null;
-            Assert.Null(option.Group);
+            Assert.Throws<ArgumentNullException>("value", () => option.Group = null);
```

Comment:
```suggestion
            AssertExtensions.Throws<ArgumentNullException>("value", () => option.Group = null);
```

Reply:
Why do we have both? They look to be duplicates to me:

https://github.com/xunit/assert.xunit/blob/80b21958285c952b210a598435bebc121bae251a/ExceptionAsserts.cs#L174-L180

```C#
        public static T Throws<T>(string paramName, Action testCode)
            where T : ArgumentException
        {
            var ex = Assert.Throws<T>(testCode);
            Assert.Equal(paramName, ex.ParamName);
            return ex;
        }
```

https://github.com/dotnet/runtime/blob/0b25063ec056b0d6668bf0848d108d8c873f3fce/src/libraries/Common/tests/CoreFx.Private.TestUtilities/System/AssertExtensions.cs#L66-L74

Reply:
For when we target a platform where size is more important than diagnostics, and we strip out resource strings, argument names, etc.  We did this with .NET Native, which is why we have this helper.  When we removed .NET Native support from dotnet/runtime, we took out the condition that was skipping the Assert.Equal check:
https://github.com/dotnet/corefx/pull/38330/files#diff-657bca01cf399192c51eaffa559337c9L39-L40
I suspect we'll find ourselves putting back that condition for other platforms at some point, like when start building the dotnet/runtime libraries to target wasm.